### PR TITLE
Minor cleanups in watch handler toward unification between http and websockets

### PR DESCRIPTION
**Pull request by wojtek-t**
```release-note
NONE
```

/kind cleanup
/priority important-longterm
/sig api-machinery

### CODE REVIEW

**Review by jpbetz — CHANGES_REQUESTED** [review]
Thanks for cleaning this up.  Happy to approver after the two (minor) comments I've added are cleared.

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go [hunk 1]
```diff
@@ -184,17 +190,9 @@ type WatchServer struct {
...
-func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
-	kind := s.Scope.Kind
-
-	if wsstream.IsWebSocketRequest(req) {
-		w.Header().Set("Content-Type", s.MediaType)
-		websocket.Handler(s.HandleWS).ServeHTTP(w, req)
-		return
-	}
-
+func (s *WatchServer) HandleHTTP(w http.ResponseWriter, req *http.Request) {
```
[the remark this answers is not in the document]
Comment (wojtek-t) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go:195]: I changed slightly what it's doing (e.g. it's not handling websocket now), so changing name helps imho [I doubt someone vendoring this repo is really using that).
But if you feel strongly about leaving the name as it was previously - I can revert it.

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go [hunk 2]
```diff
@@ -265,7 +264,7 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
-// HandleWS implements a websocket handler.
+// HandleWs serves a series of encoded events over a websocket connection.
```
[the remark this answers is not in the document]
Comment (wojtek-t) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go:267]: done

**Pull request merged**